core/src/main/scala/org/apache/spark/util/collection/CompactBuffer.scala

```diff
@@ -152,6 +152,20 @@ private[spark] class CompactBuffer[T: ClassTag] extends Seq[T] with Serializable
 }
 
 private[spark] object CompactBuffer {
+  def constantEmpty[T: ClassTag](): CompactBuffer[T] = new CompactBuffer[T] {
+    override def apply(position: Int): T =
+      sys.error("apply() is not valid for a constant empty buffer")
```

Comment:
Can this throw `UnsupportedOperationException` instead? this makes a bald `RuntimeException` which is slightly poor form to throw.